### PR TITLE
[WIP] Add Pi, Lenet, TPC-H Examples Run in PPML Trusted Big Data ML with Spark

### PR DESCRIPTION
**Pull request by ManfeiBai**

### CODE REVIEW

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md [hunk 1]
```diff
@@ -80,5 +112,103 @@ or
...
+>   ############# ModuleLoader.saveToFile File.saveBytes end, used 827002 ms[P1182:T2:java] ---- end time: 1142754 ms return from shim_write(...) = 0x48 <br>
+>   ############# ModuleLoader.saveToFile saveWeightsToFile end, used 842543 ms[P1182:T2:java] ---- end time: 1985297 ms return from shim_write(...) = 0x4b <br>
+>   ############# model saved[P1182:T2:java] ---- end time: 1985297 ms return from shim_write(...) = 0x19 <br>
+
+##### Start the container to run TPC-H in ppml.
+Download SBT: <br>
+```bash
+wget https://github.com/sbt/sbt/releases/download/v1.4.8/sbt-1.4.8.tgz
+tar -zxvf sbt-1.4.8.tgz
+wget https://archive.apache.org/dist/spark/spark-2.3.4/spark-2.3.4.tgz
```
Comment (glorysdj) [inline ppml/trusted-big-data-ml/scala/docker-graphene/README.md:126]: We already have spark in the image
ENV SPARK_HOME				/ppml/trusted-big-data-ml/work/spark-${SPARK_VERSION}
please try that spark

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md [hunk 2]
```diff
@@ -80,5 +112,103 @@ or
...
+tar -zxvf spark-2.3.4.tgz
+wget https://downloads.lightbend.com/scala/2.11.12/scala-2.11.12.tgz
+tar -zxvf scala-2.11.12.tgz
+export SBT_HOME=the_dir_path_of_your_unzipped_sbt-1.4.8
+export SPARK_HOME=the_dir_path_of_your_unzipped_spark-2.3.4
+export SCALA_HOME=the_dir_path_of_your_unzipped_scala-2.11.12
+export PATH=$SBT_HOME:$SPARK_HOME/bin:$PATH:$SCALA_HOME
```
Comment (glorysdj) [inline ppml/trusted-big-data-ml/scala/docker-graphene/README.md:133]: if we need scala and sbt, we should install it in the Dockerfile

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md [hunk 3]
```diff
@@ -50,7 +50,39 @@ The ppml in analytics zoo need secured keys to enable spark security such as AUT
...
+    -e SGX_MEM_SIZE=64G \
+    intelanalytics/analytics-zoo-ppml-trusted-big-data-ml-scala-graphene:latest \
+    bash
+sudo docker exec -it spark-local bash
+cd ppml/trusted-bid-data-ml
+export SPARK_HOME=/ppml/trusted-big-data-ml/work/spark-2.3.4
+export SCALA_HOME=/ppml/trusted-big-data-ml/work/scala-2.11.12
+export PATH=$SPARK_HOME/bin:$PATH:$SCALA_HOME
+./init.sh
+$SPARK_HOME/bin/spark-submit --class org.apache.spark.examples.SparkPi --master local $SPARK_HOME/examples/jars/spark-examples_2.11-2.4.3.jar
```
Comment (glorysdj) [inline ppml/trusted-big-data-ml/scala/docker-graphene/README.md:80]: this should run in SGX, SGX=1 ....

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md [hunk 4]
```diff
@@ -80,5 +112,103 @@ or
...
+    intelanalytics/analytics-zoo-ppml-trusted-big-data-ml-scala-graphene:latest \
+    bash
+```
+Copy TPC-H to container: <br>
+```bash
+docker cp tpch-spark/ spark-local:/ppml/trusted-big-data-ml
+sudo docker exec -it spark-local bash
+cd ppml/trusted-big-data-ml/tpch-spark
+export PATH=$SPARK_HOME/bin:$PATH
+spark-submit --class "main.scala.TpchQuery" --master local[*] target/scala-2.11/spark-tpc-h-queries_2.11-1.0.jar dbgen
```
Comment (glorysdj) [inline ppml/trusted-big-data-ml/scala/docker-graphene/README.md:203]: also need to run in SGX with SGX=1 ...

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md (hunk 3 above)
Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md:80]: > this should run in SGX, SGX=1 ....

Done

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md (hunk 1 above)
Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md:126]: > We already have spark in the image
> ENV SPARK_HOME /ppml/trusted-big-data-ml/work/spark-${SPARK_VERSION}
> please try that spark

Thanks， tried and modified~

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md (hunk 2 above)
Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md:133]: > if we need scala and sbt, we should install it in the Dockerfile

Thanks, scala and sbt would not be used inside the docker

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md (hunk 4 above)
Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md:203]: Thanks, done~

> also need to run in SGX with SGX=1 ...

Thanks, done~

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md [hunk 5]
```diff
@@ -50,11 +50,14 @@ The ppml in analytics zoo need secured keys to enable spark security such as AUT
 ### Run the PPML Docker image
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-graphene/README.md:50]: Remove `the`
Image

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md [hunk 6]
```diff
@@ -69,138 +72,128 @@ sudo docker run -itd \
     --name=spark-local \
     -e LOCAL_IP=$LOCAL_IP \
     -e SGX_MEM_SIZE=64G \
-    intelanalytics/analytics-zoo-ppml-trusted-big-data-ml-scala-graphene:latest \
+    10.239.45.10/arda/analytics-zoo-ppml-trusted-big-data-ml-scala-graphene:0.10-SNAPSHOT \
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-graphene/README.md:75]: Remove local IP `10.239.45.10`

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md [hunk 7]
```diff
@@ -69,138 +72,128 @@ sudo docker run -itd \
...
 ```
-The result shows like: <br>
->   ############# train optimized[P1182:T2:java] ---- end time: 310534 ms return from shim_write(...) = 0x1d <br>
->   ############# ModuleLoader.saveToFile File.saveBytes end, used 827002 ms[P1182:T2:java] ---- end time: 1142754 ms return from shim_write(...) = 0x48 <br>
->   ############# ModuleLoader.saveToFile saveWeightsToFile end, used 842543 ms[P1182:T2:java] ---- end time: 1985297 ms return from shim_write(...) = 0x4b <br>
->   ############# model saved[P1182:T2:java] ---- end time: 1985297 ms return from shim_write(...) = 0x19 <br>
 
-##### Start the container to run TPC-H in ppml.
-Download SBT: <br>
+Open one new screen and check the log:
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-graphene/README.md:121]: Open one new screen -> `Open another terminal`

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md [hunk 8]
```diff
@@ -69,138 +72,128 @@ sudo docker run -itd \
...
-export SPARK_HOME=the_dir_path_of_your_unzipped_spark-2.3.4
-export SCALA_HOME=the_dir_path_of_your_unzipped_scala-2.11.12
-export PATH=$SBT_HOME:$SPARK_HOME/bin:$PATH:$SCALA_HOME
-cd sbt
-vim ./sbt
+sudo docker exec -it spark-local cat /ppml/trusted-big-data-ml/spark.local.pi.sgx.log | egrep "###|INFO|Pi"
 ```
-Please add these into this sbt file and save it:
+
+The check result would shows like this: <br>
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-graphene/README.md:126]: remove check

Comment (qiyuangong) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md:126]: would look like

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md [hunk 9]
```diff
@@ -69,138 +72,128 @@ sudo docker run -itd \
...
+##### Example Test 2
 ```bash
-SBT_OPTS="-Xms512M -Xmx1536M -Xss1M -XX:+CMSClassUnloadingEnabled -XX:MaxPermSize=256M"
-java $SBT_OPTS -jar $SBT_HOME/bin/sbt-launch.jar "$@"
+./start-spark-local-train-sgx.sh
 ```
-If there is some mistake, try to change your $SBT_HOME with the absulate path
 
-Build SBT: <br>
+Open one new screen and check the log:
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-graphene/README.md:134]: Open one new screen -> `Open another terminal`

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md (hunk 5 above)
Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md:50]: > Remove `the`
> Image

Done

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md [hunk 10]
```diff
@@ -69,138 +72,128 @@ sudo docker run -itd \
...
-    --name=spark-local \
-    -e LOCAL_IP=$LOCAL_IP \
-    -e SGX_MEM_SIZE=64G \
-    intelanalytics/analytics-zoo-ppml-trusted-big-data-ml-scala-graphene:latest \
-    bash
+./start-spark-local-tpc-h-sgx.sh
 ```
-Copy TPC-H to container: <br>
+
+Open one new screen and check the log: <br>
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-graphene/README.md:194]: Open one new screen -> `Open another terminal`

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md (hunk 7 above)
Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md:121]: > Open one new screen -> `Open another terminal`

Thanks, done~

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md [hunk 11]
```diff
@@ -69,138 +72,128 @@ sudo docker run -itd \
...
 ```bash
-docker cp tpch-spark/ spark-local:/ppml/trusted-big-data-ml
-sudo docker exec -it spark-local bash
-cd ppml/trusted-big-data-ml/tpch-spark
-export PATH=$SPARK_HOME/bin:$PATH
-spark-submit --class "main.scala.TpchQuery" --master local[*] target/scala-2.11/spark-tpc-h-queries_2.11-1.0.jar dbgen
+sudo docker exec -it spark-local cat /ppml/trusted-big-data-ml/spark.local.tpc.h.sgx.log | egrep "###|INFO"
 ```
 The result shows like this: <br>
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-graphene/README.md:198]: The result should look like

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md (hunk 9 above)
Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md]: > Open one new screen -> `Open another terminal`

Thanks, done~

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md (hunk 10 above)
Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md]: > Open one new screen -> `Open another terminal`

Thanks, done~

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md (hunk 8 above)
Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md]: > remove check

Thanks, done~

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md (hunk 11 above)
Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md]: > The result should look like

Thanks, done~

**Review by qiyuangong — APPROVED** [review]
LGTM
Merge after addressing these comments.

File: ppml/trusted-big-data-ml/scala/docker-graphene/README.md (hunk 6 above)
Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md:75]: User would pull image to their local by running `sudo docker pull 10.239.45.10/arda/analytics-zoo-ppml-trusted-big-data-ml-scala-graphene:0.10-SNAPSHOT`. So we want to change local IP 10.239.45.10 back to intelanalytics/? Like the former description: intelanalytics/analytics-zoo-ppml-trusted-big-data-ml-scala-graphene:latest?

Comment (ManfeiBai) [reply ppml/trusted-big-data-ml/scala/docker-graphene/README.md]: Thanks, done~

**Pull request merged**